accessEntity/api.go

```diff
@@ -0,0 +1,721 @@
+)
+
+const (
+	AppTypeDeveloper = "DEVELOPER"
+	AppTypeCompany   = "COMPANY"
+)
+
+const (
+	StatusApproved = "APPROVED"
+	StatusRevoked  = "REVOKED"
+	StatusExpired  = "EXPIRED"
+)
+
+var (
+	Identifiers = map[string]bool{
+		"appid":          true,
+		"apiproductname": true,
+		"appname":        true,
+		"apiresource":    true,
+		"developerid":    true,
+		"companyname":    true,
+		"developeremail": true,
+		"consumerkey":    true,
+	}
+
+	ErrInvalidPar = &common.ErrorResponse{
+		ResponseCode:    strconv.Itoa(INVALID_PARAMETERS),
+		ResponseMessage: "Invalid Identifiers",
+		StatusCode:      http.StatusBadRequest,
+	}
+
+	ErrNotFound = &common.ErrorResponse{
+		ResponseCode:    strconv.Itoa(NOT_FOUND),
+		ResponseMessage: "Resource Not Found",
+		StatusCode:      http.StatusNotFound,
+	}
+
+	IdentifierTree = map[string]map[string][]string{
+		EndpointApiProduct: {
+			IdentifierApiProductName: {},
+			IdentifierAppId:          {IdentifierApiResource},
+			IdentifierAppName:        {IdentifierApiResource, IdentifierDeveloperEmail, IdentifierDeveloperId, IdentifierCompanyName},
+			IdentifierConsumerKey:    {IdentifierApiResource},
+		},
+		EndpointApp: {
+			IdentifierAppId:       {},
+			IdentifierAppName:     {IdentifierDeveloperEmail, IdentifierDeveloperId, IdentifierCompanyName},
+			IdentifierConsumerKey: {},
+		},
+		EndpointCompany: {
+			IdentifierAppId:       {},
+			IdentifierCompanyName: {},
+			IdentifierConsumerKey: {},
+		},
+		EndpointCompanyDeveloper: {
+			IdentifierCompanyName: {},
+		},
+		EndpointAppCredentials: {
+			IdentifierConsumerKey: {},
+		},
+		EndpointDeveloper: {
+			IdentifierDeveloperEmail: {},
+			IdentifierAppId:          {},
+			IdentifierDeveloperId:    {},
+			IdentifierConsumerKey:    {},
+		},
+	}
+)
+
+const (
+	INVALID_PARAMETERS = iota
+	// Server DB Error
+	DB_ERROR
+	// Invalid/Wrong Data in DB data. This probably means something wrong happened in upstream PG/Transicator.
+	DATA_ERROR
+	// 404
+	NOT_FOUND
+)
+
+type ApiManager struct {
+	DbMan            DbManagerInterface
+	AccessEntityPath string
+	apiInitialized   bool
+}
+
+func (a *ApiManager) InitAPI() {
+	if a.apiInitialized {
+		return
+	}
+	services.API().HandleFunc(a.AccessEntityPath+EndpointApp, a.HandleApps).Methods("GET")
+	services.API().HandleFunc(a.AccessEntityPath+EndpointApiProduct, a.HandleApiProducts).Methods("GET")
+	services.API().HandleFunc(a.AccessEntityPath+EndpointCompany, a.HandleCompanies).Methods("GET")
+	services.API().HandleFunc(a.AccessEntityPath+EndpointCompanyDeveloper, a.HandleCompanyDevelopers).Methods("GET")
+	services.API().HandleFunc(a.AccessEntityPath+EndpointDeveloper, a.HandleDevelopers).Methods("GET")
+	services.API().HandleFunc(a.AccessEntityPath+EndpointAppCredentials, a.HandleAppCredentials).Methods("GET")
+	a.apiInitialized = true
+	log.Debug("API endpoints initialized")
+}
+
+func (a *ApiManager) handelEndpoint(endpoint string, w http.ResponseWriter, r *http.Request) {
```

Comment:
handleEndpoint...Typo in function name. (nothing against the logic).

Reply:
Good catch!